hazelcast/src/main/java/com/hazelcast/map/impl/query/PartitionScanRunner.java

```diff
@@ -98,6 +103,49 @@ public PartitionScanRunner(MapServiceContext mapServiceContext) {
         return getSortedSubList(resultList, pagingPredicate, nearestAnchorEntry);
     }
 
+    /**
+     * Executes the predicate on a partition chunk. The offset in the partition is defined by the {@code tableIndex}
+     * and the soft limit is defined by the {@code fetchSize}. The method returns the matched entries and an
+     * index from which new entries can be fetched which allows for efficient iteration of query results.
+     * <p>
+     * <b>NOTE</b>
+     * The reading should be done when the {@link IMap} is static. When reading from a Map, with or without
```

Comment:
Nitpicking: `Static` is a quite misleading word. I would add (so `no writes are allowed`). Also I don't quite get the rest of the javadoc, I mean I don't know what it is trying to say.